_posts/2017-11-10-choosing-a-cni-provider.md

```diff
@@ -0,0 +1,315 @@
+such as BGP, egress control, and mesh networking are only available with
+different CNI providers.
+
+## CNI in kops
+
+At last count, kops supports seven different CNI providers besides kubenet.
+Choosing from seven different network providers is a daunting task.
+
+Here is our current list of providers that can be installed out of the box,
+sorted in alphabetical order.
+
+- [Calico](http://docs.projectcalico.org/)
+- [Canal (Flannel + Calico)](https://github.com/projectcalico/canal)
+- [flannel](https://github.com/coreos/flannel)
+- [kopeio-vxlan](https://github.com/kopeio/networking)
+- [kube-router](https://github.com/cloudnativelabs/kube-router)
+- [romana](https://github.com/romana/romana)
+- [weave](https://github.com/weaveworks/weave-kube)
+
+Any of these CNI providers can be used without kops. All of the CNI providers
+use a daemonset installation model, where their product deploys a Kubernetes
+Daemonset.  Just use kubectl to install the provider on the master, when the K8s
+API server has started. Please refer to each projects specific documentation.
+
+## Summary of the Providers
+
+### Calico
+
+Mike Stowe provided a summary of both Calico and Canal.
+
+Calico provides simple, scalable networking using a pure L3 approach.  It
+enables native, unencapsulated networking in environments that support it,
+including AWS AZ's and other environments with L2 adjacency between nodes, or in
+deployments where it's possible to peer with the infrastructure using BGP, such
+as on-premise. Calico also provides a stateless IP-in-IP mode that can be used
+in other environments, if necessary. Beyond scalable networking, Project Calico
+also offers policy isolation, allowing you to secure and govern your
+microservices/container infrastructure using advanced ingress and egress
+policies.   With extensive Kubernetes support, you’re able to manage your
+policies, in Kubernetes 1.8+.
+
+### Canal
+
+Canal is a CNI provider that gives you the best of Flannel and Project Calico,
+providing simple, easy to use/ out of the box VXLAN networking while also
+allowing you take advantage of policy isolation with Calico policies.
+
+This provider is a solution for anyone who wants to get up and running while
+taking advantage of familiar technologies that they may already be using.
+
+### flannel
+
+Brandon Phillips' views on flannel.
+
+Flannel is a simple and easy way to configure a layer3 network fabric designed
+for Kubernetes. No external database (uses kubernetes API), simple performant
+works anywhere VXLAN default, can be layered with Calico policy engine (Canal).
+Oh, and lots of users.
+
+Techtonics,  CoreOS's Commercial Kuberentes product, uses a combination of
+flannel and Felix from Calico, much like Canal.
+
+### kopeio-vxlan
+
+Justin Santa Barbara the founder of kope-vxlan provided this:
+
+Pioneered the model that everyone is now using, No baggage, with  Minimal CNI
+dependencies.  Currently, there is lower adoption.
```

Comment:
Ah well, if we're doing marketing speak.

"kopeio-networking provides kubernetes-first networking.  It was purpose built for Kubernetes, making full use of the Kubernetes API, and because of that is much simpler and more reliable than alternatives that were retrofitted.  The VXLAN approach is the most commonly used mode (as used in weave & flannel), but it also supports layer 2 (as used in calico), with more experimental support for GRE (the replacement for IPIP), and for IPSEC (for secure configurations).  It does all of this with a very simple codebase."